qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -280,34 +364,38 @@ def compose(self, other, qargs=None, front=False):
         other = self._pad_with_identity(other, qargs)
 
         if front:
-            table1 = self.table
-            table2 = other.table
+            clifford1 = self
+            clifford2 = other
         else:
-            table1 = other.table
-            table2 = self.table
+            clifford1 = other
+            clifford2 = self
 
         num_qubits = self.num_qubits
+        x_indices = slice(0, num_qubits)
+        z_indices = slice(num_qubits, 2 * num_qubits)
 
-        array1 = table1.array.astype(int)
-        phase1 = table1.phase.astype(int)
-
-        array2 = table2.array.astype(int)
-        phase2 = table2.phase.astype(int)
+        array1 = clifford1.paulis.tableau().astype(int)
+        phase1 = clifford1.paulis.phase
 
-        # Update Pauli table
-        pauli = StabilizerTable(array2.dot(array1) % 2)
+        array2 = clifford2.paulis.tableau().astype(int)
+        phase2 = clifford2.paulis.phase
 
-        # Add phases
-        phase = np.mod(array2.dot(phase1) + phase2, 2)
+        # Update Pauli list
+        composed_array = array2.dot(array1) % 2
+        pauli = PauliList.from_symplectic(
+            z=composed_array[:, z_indices],
+            x=composed_array[:, x_indices],
+            phase=array2.dot(phase1) + phase2,
+        )
```

Comment:
Just for future reference/other reviewers: I'm not bothered about reviewing efficiency in `Clifford.compose` because we intend to merge a PR completely rewriting it (#7483) once this deprecation is merged.